File: townsquare-backend/src/main/java/com/timesquare/services/CommentService.java
package com.timesquare.services;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.timesquare.models.Comment;
import com.timesquare.repos.CommentRepository;

@Service
public class CommentService {
	
	@Autowired
	private CommentRepository commentRepo;
	
	public List<Comment> getAllComments() {
		return commentRepo.findAll();
	}
	
	public List<Comment> getAllUserComments(Long userId) {
		return commentRepo.findAll().stream()
					.filter((comment) -> comment.getUser().getUserId() == userId)
					.collect(Collectors.toList());
	}
	
	public Comment getCommentById(Long commentId) throws Exception {
		return commentRepo.findById(commentId).orElseThrow(
				() -> new Exception("Comment not found with id " + commentId));
	}
	
	public String addComment(Comment comment) {
		commentRepo.save(comment);
		return comment.getUser().getFirstName() + " " + comment.getUser().getLastName()
				+ " commented on your post!";
	}
	
	public String updateComment(Comment comment) {
		if (commentRepo.findById(comment.getCommentId()).isPresent()) {
			commentRepo.save(comment);
			return comment.getUser().getFirstName() + " " + comment.getUser().getLastName()
					+ "'s comment was edited.";
		}
		return "No comment found with the id " + comment.getCommentId();
	}
	
	public String deleteComment(Long commentId) {
		if (commentRepo.findById(commentId).isPresent()) {
			commentRepo.deleteById(commentId);
			return "Comment deleted with id " + commentId;
		}
		return "No comment found with id " + commentId;
	}

}

File: townsquare-frontend/townsquare-fe/src/app/services/comment/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CommentService {

  constructor(private http: HttpClient) { }

  getAllComments(): Observable<Comment[]> {
    return this.http.get<Comment[]>(`http://localhost:8181/api/comment`);
  }

  getAllUserComments(userId: number): Observable<Comment[]> {
    return this.http.get<Comment[]>(`http://localhost:8181/api/comment/${userId}`);
  }

  getCommentById(commentId: number): Observable<Comment> {
    return this.http.get<Comment>(`http://localhost:8181/api/comment/id/${commentId}`);
  }

  addComment(comment: Comment): Observable<String> {
    return this.http.post<String>(`http://localhost:8181/api/comment/add`, comment);
  }

  updateComment(comment: Comment): Observable<String> {
    return this.http.put<String>(`http://localhost:8181/api/comment/update`, comment);
  }

  deleteComment(commentId: number): Observable<String> {
    return this.http.delete<String>(`http://localhost:8181/api/comment/delete/${commentId}`);
  }
  
}
